typescript/packages/common-ui/src/components/common-secret.ts

```diff
@@ -0,0 +1,191 @@
+
+  constructor(detail: CommonSecret) {
+    super("common-secret", { bubbles: true, composed: true });
+    this.detail = detail;
+  }
+}
+
+function pemToArrayBuffer(pem: string): ArrayBuffer {
+  const pemHeader = "-----BEGIN PUBLIC KEY-----";
+  const pemFooter = "-----END PUBLIC KEY-----";
+  let pemContents = pem.replace(pemHeader, "").replace(pemFooter, "");
+  pemContents = pemContents.replace(/\s/g, "");
+  const binaryString = window.atob(pemContents);
+  const len = binaryString.length;
+  const bytes = new Uint8Array(len);
+  for (let i = 0; i < len; i++) {
+    bytes[i] = binaryString.charCodeAt(i);
+  }
+  return bytes.buffer;
+}
+
+async function importPublicKey(pemKey: string): Promise<CryptoKey> {
+  const binaryDer = pemToArrayBuffer(pemKey);
+  return window.crypto.subtle.importKey(
+    "spki",
+    binaryDer,
+    {
+      name: "RSA-OAEP",
+      hash: "SHA-256",
+    },
+    true,
+    ["encrypt"],
+  );
+}
+
+// Helper: Convert an ArrayBuffer to a base64-encoded string.
+function arrayBufferToBase64(buffer: ArrayBuffer): string {
+  const bytes = new Uint8Array(buffer);
+  let binary = "";
+  for (let i = 0; i < bytes.byteLength; i++) {
+    binary += String.fromCharCode(bytes[i]);
+  }
+  return window.btoa(binary);
+}
+
+@customElement("common-secret")
+export class CommonSecretElement extends LitElement {
+  static override styles = [
+    baseStyles,
+    css`
+      :host {
+        display: block;
+        --height: 24px;
+      }
+      .input-wrapper {
+        display: flex;
+        flex-direction: column;
+        gap: 8px;
+      }
+      .input-group {
+        display: flex;
+        gap: 8px;
+        align-items: center;
+      }
+      .input {
+        appearance: none;
+        border: 1px solid var(--border-color, #ccc);
+        outline: none;
+        box-sizing: border-box;
+        font-size: var(--body-size, 14px);
+        width: 100%;
+        height: var(--height);
+        padding: 4px 8px;
+      }
+      :host([appearance="rounded"]) .input {
+        border-radius: calc(var(--height) / 2);
+      }
+      button {
+        padding: 4px 12px;
+        font-size: var(--body-size, 14px);
+        cursor: pointer;
+        transition:
+          opacity 0.2s,
+          background-color 0.2s;
+      }
+      button:disabled {
+        cursor: not-allowed;
+        opacity: 0.6;
+      }
+      .input:focus {
+        border-color: var(--focus-color, #0066cc);
+        box-shadow: 0 0 0 2px var(--focus-ring-color, rgba(0, 102, 204, 0.2));
+      }
+      .input:disabled {
+        background-color: var(--disabled-bg, #f5f5f5);
+        cursor: not-allowed;
+      }
+      .warning {
+        color: red;
+        font-size: 0.9em;
+      }
+    `,
+  ];
+
+  @property({ type: String }) value = "";
+  @property({ type: String }) placeholder = "Enter secret";
+  @property({ type: String }) appearance = "default";
+  @property({ type: String, attribute: "pubkey" }) pubkey = "";
```

Comment:
May be worthwhile mentioning that the `pubkey` field must be base64 encoded, AFAICT